sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -0,0 +1,1144 @@
+        @Nullable ContentStructure contentStructure) {
+      this.fhirStore = fhirStore.get();
+      this.deadLetterGcsPath = deadLetterGcsPath.get();
+      if (contentStructure == null) {
+        this.contentStructure = ContentStructure.CONTENT_STRUCTURE_UNSPECIFIED;
+      } else {
+        this.contentStructure = contentStructure;
+      }
+    }
+    /**
+     * Instantiates a new Import.
+     *
+     * @param fhirStore the fhir store
+     * @param tempGcsPath the temp gcs path
+     * @param deadLetterGcsPath the dead letter gcs path
+     * @param contentStructure the content structure
+     */
+    Import(
+        String fhirStore,
+        String tempGcsPath,
+        String deadLetterGcsPath,
+        @Nullable ContentStructure contentStructure) {
+      this.fhirStore = fhirStore;
+      this.tempGcsPath = tempGcsPath;
+      this.deadLetterGcsPath = deadLetterGcsPath;
+      if (contentStructure == null) {
+        this.contentStructure = ContentStructure.CONTENT_STRUCTURE_UNSPECIFIED;
+      } else {
+        this.contentStructure = contentStructure;
+      }
+    }
+
+    @Override
+    String getFhirStore() {
+      return fhirStore;
+    }
+
+    @Override
+    WriteMethod getWriteMethod() {
+      return WriteMethod.IMPORT;
+    }
+
+    @Override
+    Optional<ContentStructure> getContentStructure() {
+      return Optional.of(contentStructure);
+    }
+
+    @Override
+    Optional<String> getImportGcsTempPath() {
+      return Optional.of(tempGcsPath);
+    }
+
+    @Override
+    Optional<String> getImportGcsDeadLetterPath() {
+      return Optional.of(deadLetterGcsPath);
+    }
+
+    @Override
+    public Write.Result expand(PCollection<String> input) {
+      checkState(
+          input.isBounded() == IsBounded.BOUNDED,
+          "FhirIO.Import should only be used on unbounded PCollections as it is"
+              + "intended for batch use only.");
+
+      // fall back on pipeline's temp location.
+      String tempPath = tempGcsPath;
+      if (tempPath == null) {
+        tempPath = input.getPipeline().getOptions().getTempLocation();
+      }
+
+      // Write bundles of String to GCS
+      PCollectionTuple writeTmpFileResults =
+          input.apply(
+              "Write nd json to GCS",
+              ParDo.of(new WriteBundlesToFilesFn(fhirStore, tempPath, deadLetterGcsPath))
+                  .withOutputTags(Write.TEMP_FILES, TupleTagList.of(Write.FAILED_BODY)));
+
+      PCollection<HealthcareIOError<String>> failedBodies =
+          writeTmpFileResults
+              .get(Write.FAILED_BODY)
+              .setCoder(HealthcareIOErrorCoder.of(StringUtf8Coder.of()));
+      int numShards = 100;
+      PCollection<HealthcareIOError<String>> failedFiles =
+          writeTmpFileResults
+              .get(Write.TEMP_FILES)
+              .apply(
+                  "Shard files", // to paralelize group into batches
+                  WithKeys.of(ThreadLocalRandom.current().nextInt(0, numShards)))
```

Comment:
This is providing a static key, not a key-generating function, right?
```suggestion
                  WithKeys.of(elm -> ThreadLocalRandom.current().nextInt(0, numShards)))
```
(I think)